### ISSUE TITLE
Window open rules in wavebox 10

### ISSUE THREAD

**Issue by TheSpyder** [wavebox/waveboxapp#1185]
In v4 I could shift+click a link to customise the behaviour (open in default browser, new window, or a new window from a different profile). I can see there is still a "Window open rules" section in the app config, but it seems there's no way to manually customise it?
https://kb.wavebox.io/customise-new-window-handling/

This is a big step backwards from v4. I have come to rely heavily on the ability to customise links in two ways:

1) I could have links from slack or my email automatically open a new window from the correct app. All the convenience of a normal browser with none of the profile/login sharing. "App Matcher" behaviour sounds like it should do that, but it seems to only match within the current app.
2) Any link that doesn't match an app within wavebox must open in my default browser. This matters for cases like github, which I don't see much point adding to wavebox, but now links open to 404s for private repos which is quite disconcerting.

**Comment by Thomas101** [comment]
Hi, this is something that's changed between version 4 & 10 and we're still looking to make sure that we can cover all bases, such as those that you've mentioned above.

One of the reasons for the change is to provide fewer interruptions, by constantly bringing up popups to ask what to do - but from what it sounds like, in your 4 setup you were using it in the inverse way so that you could optionally keep it in Wavebox and keep it in the same profile.

I think you might be able to reproduce a similar behaviour in 10, but it would be great to get some feedback.

The first thing to do is set the `Shift+Link behaviour` to open in new window, and the `Opening new windows` to be the default browser

![Screenshot 2020-01-10 at 08 57 45](https://user-images.githubusercontent.com/103586/72140085-dc6db400-3387-11ea-9997-c820f1298d14.png)

This will give the behaviour of having all links open in the default browser, unless you Shift+Click, in which case they'll open in Wavebox.

The second thing is in Wavebox 4 each group down the sidebar had its own profile. This was hard set and you couldn't share profiles between apps/groups. In Wavebox 10, [profiles are a lot more](https://kb.wavebox.io/what-are-profiles-and-how-do-i-create-one-in-wavebox-sandboxing/) dynamic, you've got the option to group similar apps together in single profiles, meaning they share cookies and logins. You can see which apps use each profile under Settings > Profiles ...

![Screenshot 2020-01-10 at 09 05 38](https://user-images.githubusercontent.com/103586/72140421-a54bd280-3388-11ea-85ca-752581f72bf8.png)

... You can re-configure these under each accounts setting. What this means is that when links are opened in a Wavebox window (via shift+click in the above case) they'll open in the current profile - if apps are grouped logically into profiles sign-ins should just work

Does that cover some or any of the cases? It would be great to get some feedback so that we can try to cover off any final cases

**Comment by chrisblossom** [comment]
I think you used to be able to set it so one profile would open Chrome and another profile open Firefox (my default browser). I'd like my work email to open links in Chrome, and my personal email to open links in Firefox.

_(If this is a separate issue let me know and I'll open a new issue)_

**Comment by TheSpyder** [comment]
Sorry I've taken 3 weeks to get back to this, life has been very busy. I didn't want to half write this and risk losing it in a browser reload or restart 🙂 

I totally agree with fewer interruptions. It always seemed weird to me that you went for that big in your face dialog, although the effectiveness of the rules that came out of it more than made up for what was really just a temporary annoyance. I have come to rely heavily on what are perhaps edge cases in the rule capabilities.

Sharing profiles between apps is a great step forward, and it will let me have separate apps where previously I used weblink tabs that were weird (e.g. a CI tab in my gmail app). However I still have need of the v4 rule flexibility that let me direct certain links across apps and spit everything else out to the default browser.

* I have grown fond of sandboxing social media and chat apps off completely from everything else, not logging into any other service in the same profile. I could perhaps accept logging into my work slack account using the same profile as my other work-related apps, but in my personal accounts I would prefer not to.
* I use my personal github account for work, so I receive github emails to both work and home gmail accounts and I don't want to log into github on both profiles (I'd rather direct links from either place - and slack etc - to my default browser or perhaps a github app)
* I tried your suggestion but holding shift to open the link in the local app profile, after months of having it be automatic, is almost enough to make me go back to v4
* Holding shift doesn't work in google drive, where links open in new windows by default, and I can't right click to override

Really what I would love (and I _might_ have mentioned this when the rules went in?) is a global set of rules. If I go back to v4 and look at my rule setup for each app they will all be identical; across every app `notion.so` links go to the notion app, `google.com` links that don't come from a google app go to the work google app, all of the JIRA/Zendesk/etc etc links go to their respective apps.

**Comment by Thomas101** [comment]
@chrisblossom can you open another issue for this? It's something that we haven't migrated across just yet, but are working on adding. It might be easier to track it separately. Also good to know which OS you're using :)

@TheSpyder thanks for the detailed explanation. I think if we add the ability to create custom rules in we could make this work. As you mentioned something that lets you do...
* `notion.so` -> `notion app` etc (at the moment the matcher only automatically works across the same profile, we're working on changing this so it works across different profiles)
* `github.com` -> new window in `personal profile`
Also maybe something like using Shift+Click could bring up a rule creation dialog with some bits prefilled, just to help you get setup

**Comment by TheSpyder** [comment]
Matching across profiles would solve my major concern, but custom rules would be awesome!

That might also solve a long-standing issue I've had with the limitations of the v4 rules; if I opened for example a Zendesk link from Slack, and then inside Zendesk click a JIRA link, it tries to open that in the Zendesk profile instead of switching to the JIRA profile. This didn't _always_ happen so I hadn't logged an issue, but it was really annoying when it did 😂 

That workflow might be ok if I start combining work-related profiles, actually 🤔

**Comment by NoSync** [comment]
> Really what I would love (and I _might_ have mentioned this when the rules went in?) is a global set of rules. If I go back to v4 and look at my rule setup for each app they will all be identical; across every app `notion.so` links go to the notion app, `google.com` links that don't come from a google app go to the work google app, all of the JIRA/Zendesk/etc etc links go to their respective apps.

Precisely. The way I was working in Wavebox 4 can not currently be replicated in Wavebox 10, and it's actually quite easy to describe: in the whole environment, I want all links not covered by one of the added apps to go to the default system browser. As of now, I had to settle to open everything in Wavebox by default, and ctrl+click when I want a link to open in the default system browser.
